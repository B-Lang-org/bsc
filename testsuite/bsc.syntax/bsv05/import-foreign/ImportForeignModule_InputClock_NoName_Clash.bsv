// Test that input_clock and output_clock names are considered
// when generating implicit names (and that clashes are avoided)

// The names 1, 5, 7 will be chosen for the unnamed clocks

interface Ifc;
  interface Clock _clk__4;
  interface Clock _clk__6;
endinterface

import "BVI" MOD =
module mkMod (Ifc) ;

   default_clock no_clock ;
   no_reset ;

   input_clock _clk__2 (CLKA) <- exposeCurrentClock ;

   input_clock (CLK1) <- exposeCurrentClock ;

   input_clock _clk__3 (CLKB) <- exposeCurrentClock ;

   input_clock (CLK5) <- exposeCurrentClock ;

   output_clock _clk__4 (CLKC) ;

   input_clock (CLK7) <- exposeCurrentClock ;

   output_clock _clk__6 (CLKD) ;

endmodule

