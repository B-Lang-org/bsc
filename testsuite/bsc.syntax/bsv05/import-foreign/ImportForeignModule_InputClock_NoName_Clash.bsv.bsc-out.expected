checking package dependencies
=== parsed:
package ImportForeignModule_InputClock_NoName_Clash;
interface Ifc;
    method Clock _clk__4();
    method Clock _clk__6();
endinterface: Ifc

module mkMod(Ifc);
  let _bv4 <- exposeCurrentClock;
  let _bv3 <- exposeCurrentClock;
  let _bv2 <- exposeCurrentClock;
  let _bv1 <- exposeCurrentClock;
  let _bv0 <- exposeCurrentClock;
  Ifc _m__;
  _m__ <- liftModule((unexpected) module verilog "MOD";
		     clock _clk__4(CLKC);
		     clock _clk__6(CLKD);
		     clock _clk__2(CLKA, {-inhigh-});
		     clock _clk__1(CLK1, {-inhigh-});
		     clock _clk__3(CLKB, {-inhigh-});
		     clock _clk__5(CLK5, {-inhigh-});
		     clock _clk__7(CLK7, {-inhigh-});
			 ((ClockArg _clk__2, _bv4),
			  (ClockArg _clk__1, _bv3),
			  (ClockArg _clk__3, _bv2),
			  (ClockArg _clk__5, _bv1),
			  (ClockArg _clk__7, _bv0))
			 {
			   clock _clk__4;
			   clock _clk__6;
			 }
			 SchedInfo [] [] [] []
			 No combinational paths from inputs to outputs);
  method _clk__4() ;   return (_m__._clk__4);
  endmethod: _clk__4
  method _clk__6() ;   return (_m__._clk__6);
  endmethod: _clk__6
endmodule: mkMod

endpackage: ImportForeignModule_InputClock_NoName_Clash

-----

compiling ImportForeignModule_InputClock_NoName_Clash.bsv
=== parsed:
package ImportForeignModule_InputClock_NoName_Clash;
interface Ifc;
    method Clock _clk__4();
    method Clock _clk__6();
endinterface: Ifc

module mkMod(Ifc);
  let _bv4 <- exposeCurrentClock;
  let _bv3 <- exposeCurrentClock;
  let _bv2 <- exposeCurrentClock;
  let _bv1 <- exposeCurrentClock;
  let _bv0 <- exposeCurrentClock;
  Ifc _m__;
  _m__ <- liftModule((unexpected) module verilog "MOD";
		     clock _clk__4(CLKC);
		     clock _clk__6(CLKD);
		     clock _clk__2(CLKA, {-inhigh-});
		     clock _clk__1(CLK1, {-inhigh-});
		     clock _clk__3(CLKB, {-inhigh-});
		     clock _clk__5(CLK5, {-inhigh-});
		     clock _clk__7(CLK7, {-inhigh-});
			 ((ClockArg _clk__2, _bv4),
			  (ClockArg _clk__1, _bv3),
			  (ClockArg _clk__3, _bv2),
			  (ClockArg _clk__5, _bv1),
			  (ClockArg _clk__7, _bv0))
			 {
			   clock _clk__4;
			   clock _clk__6;
			 }
			 SchedInfo [] [] [] []
			 No combinational paths from inputs to outputs);
  method _clk__4() ;   return (_m__._clk__4);
  endmethod: _clk__4
  method _clk__6() ;   return (_m__._clk__6);
  endmethod: _clk__6
endmodule: mkMod

endpackage: ImportForeignModule_InputClock_NoName_Clash

-----

All packages are up to date.
